core/client/src/main/java/alluxio/client/file/policy/LocalFirstAvoidEvictionPolicy.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.client.file.policy;
+
+import alluxio.Configuration;
+import alluxio.PropertyKey;
+import alluxio.client.block.BlockWorkerInfo;
+import alluxio.util.network.NetworkAddressUtils;
+import alluxio.wire.WorkerNetAddress;
+
+import com.google.common.base.Objects;
+import com.google.common.collect.Lists;
+
+import java.util.Collections;
+import java.util.List;
+
+import javax.annotation.concurrent.ThreadSafe;
+/**
+ * A policy that returns local host first, and if the local worker doesn't have enough availability,
+ * it randomly picks a worker from the active workers list for each block write.
+ * If No worker meets the demands, return local host.
+ * USER_FILE_WRITE_AVOID_EVICTION_POLICY_RESERVED_BYTES is used to reserve some space of the worker
+ * to store the block, for the values mCapacityBytes minus mUsedBytes is not the available bytes.
+ */
+@ThreadSafe
+public final class LocalFirstAvoidEvictionPolicy implements FileWriteLocationPolicy {
+  private String mLocalHostName;
+  /**
+   * Constructs a {@link LocalFirstAvoidEvictionPolicy}.
+   */
+  public LocalFirstAvoidEvictionPolicy() {
+    mLocalHostName = NetworkAddressUtils.getLocalHostName();
+  }
+
+  @Override
+  public WorkerNetAddress getWorkerForNextBlock(Iterable<BlockWorkerInfo> workerInfoList,
+      long blockSizeBytes) {
+    // try the local host first
+    WorkerNetAddress localWorkerNetAddress = null;
+    for (BlockWorkerInfo workerInfo : workerInfoList) {
+      if (workerInfo.getNetAddress().getHost().equals(mLocalHostName)) {
+        localWorkerNetAddress = workerInfo.getNetAddress();
+        if (getAvailableBytes(workerInfo) >= blockSizeBytes) {
+          return localWorkerNetAddress;
+        }
+      }
+    }
+
+    // otherwise randomly pick a worker that has enough availability
+    List<BlockWorkerInfo> shuffledWorkers = Lists.newArrayList(workerInfoList);
+    Collections.shuffle(shuffledWorkers);
+    for (BlockWorkerInfo workerInfo : shuffledWorkers) {
+      if (getAvailableBytes(workerInfo) >= blockSizeBytes) {
+        return workerInfo.getNetAddress();
+      }
+    }
+    return localWorkerNetAddress;
```

Comment:
If no worker on the local host, this policy will choice a random worker that have available capacity to store the block. If on worker have spare capacity, it should return null. This policy is similar to LocalFirstPolicy, but if local host do not have the spare capacity, it will select other worker instead of evict old block。